packages/graphiql/src/components/DocExplorer.js

```diff
@@ -142,6 +168,24 @@ export class DocExplorer extends React.Component {
 
   // Public API
   showDoc(typeOrField) {
+    if (!typeOrField) { return; }
+
+    if (this.props.onShowDoc) {
+      let typeName;
+      let fieldName;
+      if (isType(typeOrField)) {
+        typeName = typeOrField.name;
+        fieldName = '';
+      } else {
```

Comment:
I'd like to see an `else if (isField(typeOrField))` here, since we may add other entities such as directives to the `showDoc`.